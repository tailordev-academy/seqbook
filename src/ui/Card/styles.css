.Card {
  text-align: center;
}

.Card-title {
  text-transform: uppercase;
}

.Card-value {
  font-size: 3em;
}
