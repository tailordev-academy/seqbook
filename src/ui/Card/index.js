import React from 'react';
import PropTypes from 'prop-types';

import './styles.css';

const Card = props =>
  <div className="Card panel panel-default">
    <div className="Card-title panel-heading">
      <span className="panel-title">
        {props.title}
      </span>
    </div>
    <div className="Card-value panel-body">
      {props.value}
      {props.unit || ''}
    </div>
  </div>;

Card.propTypes = {
  title: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  unit: PropTypes.string,
};

export default Card;
