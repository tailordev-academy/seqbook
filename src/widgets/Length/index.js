/* @flow */
import React from 'react';

import Card from 'ui/Card';

type Props = {|
  dna: string,
|};

const Length = ({ dna }: Props) => <Card title="length" value={dna.length} />;

export default Length;
