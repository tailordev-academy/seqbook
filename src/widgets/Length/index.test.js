import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Card from 'ui/Card';

import Length from './index';

it('renders correctly', () => {
  const sequence = 'ATCG';
  const tree = renderer.create(<Length sequence={sequence} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('uses a Card', () => {
  const wrapper = shallow(<Length sequence={''} />);
  expect(wrapper.find(Card)).toHaveLength(1);
});
