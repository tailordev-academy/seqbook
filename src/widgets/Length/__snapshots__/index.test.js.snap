// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Card panel panel-default"
>
  <div
    className="Card-title panel-heading"
  >
    <span
      className="panel-title"
    >
      length
    </span>
  </div>
  <div
    className="Card-value panel-body"
  >
    4
    
  </div>
</div>
`;
